packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,252 @@
+  border-radius: 3px;
+  box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
+  position: relative;
+  pointer-events: all;
+  transform: translate3d(0, -16px, 0);
+  opacity: 0;
+`;
+
+const visibleModalContentStyle = css`
+  transform: translate3d(0, 0, 0);
+  opacity: 1;
+`;
+
+const modalSizes: { readonly [K in ModalSize]: string } = {
+  small: css`
+    width: 400px;
+  `,
+
+  default: css`
+    width: 720px;
+  `,
+
+  large: css`
+    ${mq({
+      width: ['720px', '720px', '960px'],
+    })}
+  `,
+};
+
+const closeButton = css`
+  color: ${uiColors.gray.dark1};
+  position: absolute;
+  cursor: pointer;
+  right: 12px;
+  top: 12px;
+`;
+
+const escapeKey = 27;
+
+interface ModalProps {
+  /**
+   * Content that will appear inside of the Modal component.
+   */
+  children: React.ReactNode;
+
+  /**
+   * Determines the open state of the modal
+   *
+   * default: `false`
+   */
+  open: boolean;
+
+  /**
+   * Specifies the size of the Modal.
+   *
+   * default: `default`
+   */
+  size?: ModalSize;
+
+  /**
+   * Callback to change the open state of the Modal.
+   *
+   */
+  setOpen?: (open: boolean) => void | React.Dispatch<SetStateAction<boolean>>;
+
+  /**
+   * Callback to determine whether or not Modal should close when user tries to close it.
+   *
+   */
+  shouldClose?: () => boolean;
+
+  /**
+   * className applied to root overlay div.
+   *
+   */
+  className?: string;
+}
+
+/**
+ * # Modal
+ *
+ *  Modals place content on top of main window.
+ *
+```
+<Modal
+  open
+  size="large"
+  setOpen={setOpen}
+  shouldClose={() => console.log('Modal is closing now!')}
+  >
+  Modal content!
+</Modal>
+```
+ * @param props.open Boolean to describe whether or not Modal is open.
+ * @param props.size String to determine size of Modal. ['small', 'default', 'large']
+ * @param props.setOpen Callback to change the open state of Modal.
+ * @param props.children Content to appear inside of Modal container.
+ * @param props.shouldClose Callback to determine whether or not Modal should close when user tries to close it.
+ * @param props.className className applied to overlay div.
+ *
+ */
+function Modal({
+  open = false,
+  size = ModalSize.Default,
+  setOpen = () => {},
+  shouldClose = () => true,
+  children,
+  className,
+  ...rest
+}: ModalProps) {
+  const scrollContainerRef = useRef<HTMLDivElement>(null);
+
+  const handleClose = useCallback(() => {
+    if (setOpen && shouldClose()) {
+      setOpen(false);
+    }
+  }, [setOpen, shouldClose]);
+
+  const handleBackdropClick = (e: React.SyntheticEvent) => {
+    if (scrollContainerRef.current && e.target === scrollContainerRef.current) {
+      handleClose();
+    }
+  };
+
+  const handleEscape = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.keyCode === escapeKey) {
+        handleClose();
+      }
+    },
+    [handleClose],
+  );
+
+  useEventListener('keydown', handleEscape);
+
+  return (
+    <Transition in={open} timeout={500} mountOnEnter unmountOnExit>
```

Comment:
I'm not seeing any transition when running this locally.

Reply:
I'm seeing it locally... I can send a video to the channel!